src/basic/fs-util.h

```diff
@@ -78,6 +78,7 @@ enum {
         CHASE_OPEN        = 1U << 4,   /* If set, return an O_PATH object to the final component */
         CHASE_TRAIL_SLASH = 1U << 5,   /* If set, any trailing slash will be preserved */
         CHASE_STEP        = 1U << 6,   /* If set, just execute a single step of the normalization */
+        CHASE_NOFOLLOW    = 1U << 7,   /* If set, don't follow the final component if it's a symlink */
```

Comment:
hmm, this should be clarified a bit. if we have a symlink change "a → b → [non-existing]", then resolving "a" would return "a" rather than "b", hence it's final only in the sense that it is the last part of the path, not the last part of the chain